### PR TITLE
[MM-56006] Convert ./components/discard_changes_modal.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Alpha-4**
**Summary**
Fixes https://github.com/mattermost/mattermost/issues/25598

**Ticket Link**
JIRA LINK: https://mattermost.atlassian.net/browse/MM-56006

**QA Notes**
Component is converted to React Functional Component, no change in functionality is expected. Component appears on Admin Console.

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/discard_changes_modal.tsx [hunk 1]
```diff
@@ -12,24 +12,23 @@ type Props = {
     onCancel: (checked: boolean) => void;
 }
 
-export default class DiscardChangesModal extends React.PureComponent<Props> {
-    public render(): JSX.Element {
-        const title: JSX.Element = (
+const DiscardChangesModal = ({show, onConfirm, onCancel}: Props) => {
```
Comment (larkox) [inline webapp/channels/src/components/discard_changes_modal.tsx:15]: Just to follow the pattern we follow in other files, put each argument in a different line:

```suggestion
const DiscardChangesModal = ({
    show,
    onConfirm,
    onCancel,
}: Props) => {
```

This makes it easier in the future to recognize in a glance what props have been added / removed by a change.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one minor style thing.

**Review by yasserfaraazkhan — APPROVED** [review]
tested in console by changing teams, channels data.
The discard changes works as expected.

**Comment by larkox** [comment]
Thank you for the contribution!

**Pull request merged**